slab/sound.py

```diff
@@ -563,6 +563,8 @@ def write(self, filename, normalise=False):
 		If the normalise keyword is set to True, the amplitude of the sound will be
 		normalised to 1.
 		'''
+		if bool(self.samplerate%1):
+			self = self.resample(int(self.samplerate))
```

Comment:
It may be better to check at what points non-integer sample rates are actually a problem, rather than silently converting them (will cause problems when modifying stimuli for TDT devices).